### PR TITLE
SPM-1432: fix empty strings in schema

### PR DESCRIPTION
**Pull request by MichaelMraka**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by MichaelMraka** [comment]
supersedes #951

**Comment by psegedy** [comment]
please rebase it since it conflicts with already merged #978
```
11:00:13 evaluator/evaluate.go:178:22: invalid operation: system.VmaasJSON == "" (mismatched types *string and untyped string)
```

File: database_admin/schema/create_schema.sql [hunk 1]
```diff
@@ -753,19 +753,15 @@ ON CONFLICT DO NOTHING;
...
-    synopsis         TEXT                     NOT NULL,
-    CHECK (NOT empty(synopsis)),
-    summary          TEXT                     NOT NULL,
-    CHECK (NOT empty(summary)),
-    solution         TEXT                     NOT NULL,
+    name             TEXT                     NOT NULL CHECK (NOT empty(name)),
+    description      TEXT                     NOT NULL CHECK (NOT empty(description)),
+    synopsis         TEXT                     NOT NULL CHECK (NOT empty(synopsis)),
+    summary          TEXT                     NOT NULL CHECK (NOT empty(summary)),
+    solution         TEXT                     NOT NULL CHECK (NOT empty(solution)),
```
Comment (psegedy) [inline database_admin/schema/create_schema.sql:760]:
```
time="2022-06-20T17:19:23Z" level=fatal msg="vmaas data sync failed" err="Failed to sync advisories: Erratas page download and process failed: Storing advisories: Storing advisories: ERROR: new row for relation \"advisory_metadata\" violates check constraint \"advisory_metadata_solution_check\" (SQLSTATE 23514)"
```
either there is a bug in testing data and solution is empty or it is allowed to have empty solution for an erratum

Comment (psegedy) [reply database_admin/schema/create_schema.sql:760]: `solution` is not present in epel errata - https://mirror.karneval.cz/pub/linux/fedora/epel/8/Everything/x86_64/repodata/e398dc76062bbd82f69ce299da691f254f13c41d4b93df68a6a4effc8dbee385-updateinfo.xml.bz2

**Comment by psegedy** [comment]
hm I still see an error with `solution` in logs

>```time="2022-06-22T15:18:25Z" level=fatal msg="vmaas data sync failed" err="Failed to sync advisories: Erratas page download and process failed: Storing advisories: Storing advisories: ERROR: new row for relation \"advisory_metadata\" violates check constraint \"advisory_metadata_solution_check\" (SQLSTATE 23514)"```

**Review by psegedy — CHANGES_REQUESTED** [review]

File: base/models/models.go [hunk 2]
```diff
@@ -152,7 +152,7 @@ type AdvisoryMetadata struct {
 	Description     string
 	Synopsis        string
 	Summary         string
-	Solution        string
+	Solution        *string
```
Comment (psegedy) [inline base/models/models.go:155]: maybe the `gorm:"default:null"` should be here  😄

**Pull request merged**